web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx

```diff
@@ -45,95 +44,81 @@ const setupConnectProps = {
   showHintTimeout: oneDay,
 };
 
-initialize();
-
 export default {
   title: 'Teleport/Discover/ConnectMyComputer/SetupConnect',
-  loaders: [mswLoader],
 };
 
-const noNodesHandler = rest.get(cfg.api.nodesPath, (req, res, ctx) =>
-  res(ctx.json({ items: [] }))
-);
+export const macOS = () => {
+  const ctx = createTeleportContext();
+  ctx.nodeService.fetchNodes = () => Promise.resolve({ agents: [] });
 
-export const macOS = () => (
-  <OverrideUserAgent userAgent={UserAgent.macOS}>
-    <Provider>
-      <SetupConnect {...setupConnectProps} />
-    </Provider>
-  </OverrideUserAgent>
-);
+  return (
+    <OverrideUserAgent userAgent={UserAgent.macOS}>
+      <Provider ctx={ctx}>
+        <SetupConnect {...setupConnectProps} />
+      </Provider>
+    </OverrideUserAgent>
+  );
+};
 
-macOS.parameters = {
-  msw: {
-    handlers: [noNodesHandler],
-  },
+export const Linux = () => {
+  const ctx = createTeleportContext();
+  ctx.nodeService.fetchNodes = () => Promise.resolve({ agents: [] });
+  return (
+    <OverrideUserAgent userAgent={UserAgent.Linux}>
+      <Provider ctx={ctx}>
+        <SetupConnect {...setupConnectProps} />
+      </Provider>
+    </OverrideUserAgent>
+  );
 };
 
-export const Linux = () => (
-  <OverrideUserAgent userAgent={UserAgent.Linux}>
-    <Provider>
+export const Polling = () => {
+  const ctx = createTeleportContext();
+  ctx.nodeService.fetchNodes = () => Promise.resolve({ agents: [] });
+  return (
+    <Provider ctx={ctx}>
       <SetupConnect {...setupConnectProps} />
     </Provider>
-  </OverrideUserAgent>
-);
-
-Linux.parameters = {
-  msw: {
-    handlers: [noNodesHandler],
-  },
+  );
 };
 
-export const Polling = () => (
-  <Provider>
-    <SetupConnect {...setupConnectProps} />
-  </Provider>
-);
-
-Polling.parameters = {
-  msw: {
-    handlers: [noNodesHandler],
-  },
+export const PollingSuccess = () => {
+  const ctx = createTeleportContext();
+  let called = 0;
+  ctx.nodeService.fetchNodes = () => {
```

Comment:
The previous version works just fine after removing the query params from path and adding `{ once: true }`.

<details>
<summary>Code</summary>

```typescript
const noNodesHandler = http.get(cfg.api.nodesPath, () =>
  HttpResponse.json({ items: [] })
);

export const Polling = () => {
  return (
    <Provider>
      <SetupConnect {...setupConnectProps} />
    </Provider>
  );
};

Polling.parameters = {
  msw: {
    handlers: [noNodesHandler],
  },
};

export const PollingSuccess = () => {
  return (
    <Provider>
      <SetupConnect {...setupConnectProps} pingInterval={5} />
    </Provider>
  );
};

PollingSuccess.parameters = {
  msw: {
    handlers: [
      http.get(
        cfg.api.nodesPath.split('?')[0],
        () => {
          return HttpResponse.json({ items: [] });
        },
        { once: true }
      ),
      http.get(cfg.api.nodesPath.split('?')[0], () => {
        return HttpResponse.json({ items: [{ id: '1234', hostname: 'foo' }] });
      }),
    ],
  },
};

```
</details>

Reply:
@kimlisa Could you bring back the previous versions of those calls? I believe we can keep using msw instead of overriding functions if we remove the query params.

Reply:
done: https://github.com/gravitational/teleport/pull/44041/commits/c8bc4fc7c201abb415f2098cdac873907b7c6891

Reply:
Ah sorry, I didn't mean only that story. In abb94e4d30355, there's a bunch of other stories that got those style of mocks, like `web/packages/teleport/src/Discover/Database/DeployService/ManualDeploy/ManualDeploy.story.tsx` or `web/packages/teleport/src/Discover/Kubernetes/HelmChart/HelmChart.story.tsx` – couldn't those be replaced with msw as well?

Reply:
right, sorry, those were probably after i realized that query was the issue 😅 , i'll get to those tomorrow

Reply:
@ravicious https://github.com/gravitational/teleport/pull/44041/commits/619fdf2a94661b2ad68f6f540ddc43bb19ff8515, places where we used both msw and context mocking, i replaced context mocking with msw. stories that just used context mock i kept it the same